http/logical.go

```diff
@@ -95,6 +90,9 @@ func buildLogicalRequest(core *vault.Core, w http.ResponseWriter, r *http.Reques
 
 func handleLogical(core *vault.Core, dataOnly bool, prepareRequestCallback PrepareRequestFunc) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Limit size
+		r.Body = http.MaxBytesReader(w, r.Body, MaxRequestSize)
```

Comment:
MaxBytesReader will reject the http request with a 400 bad request and properly close the connection. See https://golang.org/pkg/net/http/#MaxBytesReader